lib/heroku/helpers/addons/resolve.rb

```diff
@@ -85,16 +82,23 @@ def resolve_addon(identifier, &filter)
           if identifier =~ RESOURCE
             name = $1
 
-            addon = begin
-              get_addon(name)
+            begin
+              addon = get_addon(name)
+              return [addon] if addon
             rescue Heroku::API::Errors::Forbidden
               # treat permission error as no match because there might exist a
               # resource on someone else's app that has a name which
               # corresponds to a service name that we wish to check below (e.g.
               # "memcachier")
+            rescue Heroku::API::Errors::RequestFailed => e
+              error_id = (json_decode(e.response.body.to_s) || {})["id"]
+              if error_id == "multiple_matches"
+                # ignore; name-based matches are never ambiguous, so this
+                # identifier is probably a service/plan
```

Comment:
A-ha, thanks for explaining. That should do for now (and #1706 will help when it lands) :+1: